components/automate-cs-ocid/habitat/config/tasks/oauth_application.rake

```diff
@@ -0,0 +1,37 @@
+require 'active_record/base'
+require 'fileutils'
+namespace :oauth_application do
+  desc "Task to register oauth applications"
+  task :register => :environment do
+    # This task expects an environment variable to be set with a json string
+    # containing a list of oauth application objects. The list should have the following schema:
+    # [{"name"=>"<app name>", "redirect_uri"=>"<callback url for redirect>"}]
+    oauth_applications_json = ENV['OAUTH_APPLICATIONS_JSON']
+    oauth_applications = JSON.parse(oauth_applications_json)
+    ActiveRecord::Base.transaction do
```

Comment:
Make sure a table level lock is imposed while running this transaction so that other transactions attempting to write to the same table don't interfere with this one.

Reply:
Using `share` lock in a transaction now.